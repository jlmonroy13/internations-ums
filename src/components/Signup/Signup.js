import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import AccountFormField from '../AccountFormField/';

class Signup extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      confirmPassword: '',
      errorMessage: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(event) {
    event.preventDefault();
    const { email, password, confirmPassword } = this.state;
    if (password !== confirmPassword) {
      this.setState(() => ({
        errorMessage: 'Passwords do not match.',
      }));
    } else {
      this.props.createAccount({ email, password }).then((response) => {
        if (response.message) {
          this.setState(() => ({
            errorMessage: response.message,
          }));
        }
      });
    }
  }
  handleChange(event) {
    const { name, value } = event.target;
    this.setState(() => ({
      [name]: value,
      errorMessage: '',
    }));
  }
  render() {
    const { email, password, confirmPassword, errorMessage } = this.state;
    return (
      <div className="grid__item medium--one-half">
        <section className="account-box">
          <h1 className="account-box__title">Sign up</h1>
          <form onSubmit={event => this.handleSubmit(event)}>
            <div className="push-triple--bottom relative">
              <AccountFormField
                placeholder="Email"
                id="email"
                name="email"
                value={email}
                iconName="user"
                type="email"
                required={true}
                onChange={event => this.handleChange(event)}
              />
              <AccountFormField
                placeholder="Password"
                id="password"
                name="password"
                value={password}
                iconName="lock"
                type="password"
                required={true}
                onChange={event => this.handleChange(event)}
              />
              <AccountFormField
                placeholder="Confirm password"
                id="confirmPassword"
                name="confirmPassword"
                value={confirmPassword}
                iconName="lock"
                type="password"
                required={true}
                onChange={event => this.handleChange(event)}
              />
              <div className="account-box__error-message">{errorMessage}</div>
            </div>
            <button className="button button--submit" type="submit">
              Create Account
            </button>
          </form>
        </section>
        <div className="account__message">
          Already have an account? <Link to="/">Log in</Link>
        </div>
      </div>
    );
  }
}

Signup.propTypes = {
  createAccount: PropTypes.func.isRequired,
};

export default Signup;
