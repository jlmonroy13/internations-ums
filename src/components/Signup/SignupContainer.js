import { connect } from 'react-redux';
import { createAccount } from '../../actions/account';
import Signup from './Signup';

function mapDispatchToProps(dispatch) {
  return {
    createAccount: data => dispatch(createAccount(data)),
  };
}

export default connect(null, mapDispatchToProps)(Signup);
