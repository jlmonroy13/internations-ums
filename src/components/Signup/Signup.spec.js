import React from 'react';
import renderer from 'react-test-renderer';
import Signup from './Signup';

describe('<Signup />', () => {
  it('should render the sign up component', () => {
    const createAccount = jest.fn();
    const wrapper = renderer.create(<Signup createAccount={createAccount} />);
    expect(wrapper).toMatchSnapshot();
  });
});
