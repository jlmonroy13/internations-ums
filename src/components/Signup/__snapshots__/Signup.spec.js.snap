// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should render the sign up component 1`] = `
<div
  className="grid__item medium--one-half"
>
  <section
    className="account-box"
  >
    <h1
      className="account-box__title"
    >
      Sign up
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="push-triple--bottom relative"
      >
        <div
          className="account__form-group"
        >
          <label
            className="account__label account__label--with-icon"
            htmlFor="email"
          >
            <span
              className="icon icon--user"
            />
          </label>
          <input
            className="account__form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="account__form-group"
        >
          <label
            className="account__label account__label--with-icon"
            htmlFor="password"
          >
            <span
              className="icon icon--lock"
            />
          </label>
          <input
            className="account__form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="account__form-group"
        >
          <label
            className="account__label account__label--with-icon"
            htmlFor="confirmPassword"
          >
            <span
              className="icon icon--lock"
            />
          </label>
          <input
            className="account__form-control"
            id="confirmPassword"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="account-box__error-message"
        >
          
        </div>
      </div>
      <button
        className="button button--submit"
        type="submit"
      >
        Create Account
      </button>
    </form>
  </section>
  <div
    className="account__message"
  >
    Already have an account? 
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Log in
    </a>
  </div>
</div>
`;
